runtime/parachains/src/configuration.rs

```diff
@@ -1163,6 +1170,21 @@ pub mod pallet {
 				config.async_backing_params = new;
 			})
 		}
+
+		/// Set executor parameters for the next session
+		#[pallet::call_index(46)]
+		#[pallet::weight((
+			Weight::from_parts(0, 0), // FIXME: Benchmark weights
+			DispatchClass::Operational,
+		))]
+		pub fn set_next_session_executor_params(
```

Comment:
As mentioned, I am not really a fan of breaking consistency here. If we want to get rid of that god object, we should have some over arching plan for it. Right now we just have a god object and some random stuff extra, which even has different semantics when it comes to the timing of taking effect. I would just put it in the god object and leave splitting that one up for another PR.

Reply:
A good point! The whole thing should be refactored first before we'll be able to tear it apart.
Then the next question is, what part of `HostConfiguration` does it belong to? We have three:
* The parameters that are required for the parachains;
* The parameters that are not essential, but still may be of interest for parachains;
* Parameters that will unlikely be needed by parachains.

I bet on the second one, as we were discussing the collators would need it. That discussion is somewhat stalled, but that's still a concern.

Reply:
Looking at how current values are grouped, it feels a bit arbitrary. E.g. max code size is mandatory, but max pov size is not?

Reply:
> Looking at how current values are grouped, it feels a bit arbitrary. E.g. max code size is mandatory, but max pov size is not?

Grouping basically means that parachains can decode the [top](https://github.com/paritytech/polkadot/blob/26ff2cc11a6d9f7776c5ed7376f89c37fd214fbe/primitives/src/v4/mod.rs#L1062-L1066) configuration. Groups 2 and 3 don't really make difference to me. For now only 1 cannot be changed without special treatment.

And max pov size is available through validation data.